tests/integration_tests/functional/test_concurrency.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Tests  multiple microVMs work correctly when spawned simultaneously."""
+
+import time
+
+from framework import decorators
+
+import host_tools.network as net_tools
+
+NO_OF_MICROVMS = 20
+
+
+@decorators.test_context('ssh', NO_OF_MICROVMS)
+def test_run_concurrency(test_multiple_microvms, network_config):
+    """Check we can spawn multiple microvms."""
+    microvms = test_multiple_microvms
+    assert microvms
+    assert len(microvms) == NO_OF_MICROVMS
+
+    for i in range(NO_OF_MICROVMS):
+        microvm = microvms[i]
+        _ = _configure_and_run(microvm, {
+            "config": network_config, "iface_id": str(i)
+        })
+        # We check that the vm is running by sshing into it.
+        ssh_connection = net_tools.SSHConnection(microvm.ssh_config)
+        ssh_connection.execute_command('ls')
```

Comment:
I think we should use here the returned values and check that `stderr` is indeed empty. From what I tried if we pass an invalid command to `execute_command()`, the test in its current form won't fail :(.

Reply:
I am rather trying to test whether `execute_command` times out (i.e whether the guest is up and running and services were started). I am not interested in the output of the command or whether it succeeded. However, I see now that SSHConnection alone will try to connect to the instance and so it will fail at SSHConnection if the instance is unresponsive. I deleted the execute_command part. Take a look.

Reply:
Yep, makes sense, I would've kept that check though as it seemed like a useful double-check and it also brought some clarity. But I'm fine either way.